src/platform/silabs/DiagnosticDataProviderImpl.cpp

```diff
@@ -336,26 +336,27 @@ void DiagnosticDataProviderImpl::ReleaseNetworkInterfaces(NetworkInterface * net
 CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiBssId(MutableByteSpan & BssId)
 {
     constexpr size_t bssIdSize = 6;
+    wfx_wifi_scan_result_t ap  = { 0 };
+
     VerifyOrReturnError(BssId.size() >= bssIdSize, CHIP_ERROR_BUFFER_TOO_SMALL);
 
-    wfx_wifi_scan_result_t ap;
-    int32_t err = wfx_get_ap_info(&ap);
-    if (err == 0)
+    if (GetAccessPointInfo(ap) == CHIP_NO_ERROR)
     {
         memcpy(BssId.data(), ap.bssid, bssIdSize);
         BssId.reduce_size(bssIdSize);
         return CHIP_NO_ERROR;
     }
+
     return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
```

Comment:
as discussed, we should change this. we need to return a value

Reply:
Created a follow ticket to review all these since the return values are incorrect.